include/can_interface.h

```diff
@@ -270,7 +359,8 @@ class CANSignal : public ITypedCANSignal<SignalType>
             *reinterpret_cast<underlying_type *>(temp_buffer_ptr) = *buffer & mask;
             std::reverse(std::begin(temp_buffer), std::end(temp_buffer));
             this->signal_ = static_cast<SignalType>(
-                (((*reinterpret_cast<underlying_type *>(temp_buffer_ptr)) << position >> (64 - length))
+                (static_cast<float>((*reinterpret_cast<underlying_type *>(temp_buffer_ptr)) << position
```

Comment:
why the addition of casting to float here?

Reply:
This is to force floating point math when multiplying by CANTemplateGetFloat(factor)